sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Keys/src/AzureDataProtectionKeyVaultKeyBuilderExtensions.cs

```diff
@@ -54,5 +55,58 @@ public static IDataProtectionBuilder ProtectKeysWithAzureKeyVault(this IDataProt
 
             return builder;
         }
+
+        /// <summary>
+        /// Configures the data protection system to protect keys with specified key in Azure KeyVault.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="keyResolverFactory">The factory delegate to create the <see cref="IKeyEncryptionKeyResolver"/> to use for Key Vault access.</param>
+        /// <param name="keyIdentifier">The Azure Key Vault key identifier used for key encryption.</param>
+        /// <returns>The value <paramref name="builder"/>.</returns>
+        public static IDataProtectionBuilder ProtectKeysWithAzureKeyVault(this IDataProtectionBuilder builder, string keyIdentifier, Func<IServiceProvider, IKeyEncryptionKeyResolver> keyResolverFactory)
+        {
+            Argument.AssertNotNull(builder, nameof(builder));
+            Argument.AssertNotNull(keyResolverFactory, nameof(keyResolverFactory));
+            Argument.AssertNotNullOrEmpty(keyIdentifier, nameof(keyIdentifier));
+
+            builder.Services.AddSingleton<IActivator, DecryptorTypeForwardingActivator>();
+
+            builder.Services.AddSingleton(sp =>
+            {
+                var keyResolver = keyResolverFactory(sp);
+                return new AzureKeyVaultXmlEncryptor(keyResolver, keyIdentifier);
+            });
+
+            builder.Services.AddSingleton<IConfigureOptions<KeyManagementOptions>, ConfigureKeyManagementKeyVaultEncryptorClientOptions>();
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the data protection system to protect keys with specified key in Azure KeyVault.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="tokenCredentialFactory">The factory delegate to create the <see cref="TokenCredential"/> to use for authenticating Key Vault access.</param>
```

Comment:
swapped docs ordering

Reply:
That was me copying and pasting from the existing method. I've reordered that one too!